コンパイル時の警告の解消

- コンパイル時に以下の警告が出るため、これを解消する。ただし、これがWIndows等の環境でのコンパイルに影響が出ないようにすること。

```
❯ make check
gcc -Wall -Wextra -O2 -std=c99 -o cube cube.c -lm
cube.c: In function ‘gameLoop’:
cube.c:434:9: warning: implicit declaration of function ‘usleep’; did you mean ‘sleep’? [-Wimplicit-function-declaration]
  434 |         usleep(12000);  // 約83FPS（12ms待機）- 更に滑らかに
      |         ^~~~~~
      |         sleep
```

--- a/cube.c
+++ b/cube.c
@@ -1,20 +1,28 @@
 /* 3D回転キューブのレンダリングプログラム */
+
+#define _GNU_SOURCE             // GNU拡張を有効化（usleepのプロトタイプ宣言用）
+
+// プラットフォーム固有のヘッダー
+#ifdef _WIN32
+#include <windows.h>   // Windows API
+#include <conio.h>     // コンソール入出力
+
+// Windows用のusleep関数実装
+static inline void usleep(unsigned int usec) {
+    Sleep(usec / 1000);
+}
+#define kbhit() _kbhit()               // kbhitをWindowsの_kbhitに置換
+#else
+#include <unistd.h>   // usleepなど
+#include <termios.h>  // 端末制御
+#include <fcntl.h>    // ファイル制御
+#endif
+
+// 標準ライブラリ
 #include <math.h>      // 数学関数（sin, cosなど）
 #include <stdio.h>     // 標準入出力
 #include <string.h>    // 文字列操作
 #include <stdlib.h>    // メモリ管理
-
-#ifdef _WIN32
-// Windows用のinclude
-#include <windows.h>   // Windows API
-#include <conio.h>     // コンソール入出力
-#define usleep(x) Sleep((x) / 1000)  // usleepをWindowsのSleepに置換
-#define kbhit() _kbhit()               // kbhitをWindowsの_kbhitに置換
-#else
-// Unix/Linux用のinclude
-#include <unistd.h>   // usleepなど
-#include <termios.h>  // 端末制御
-#include <fcntl.h>    // ファイル制御
 
 // Unix/Linux用のkbhit関数実装
 static int kbhit(void) {
@@ -47,7 +55,6 @@ static int kbhit(void) {
     }
     return 0;
 }
-#endif
 
 // デフォルト設定値の定義
 #define DEFAULT_WIDTH 160           // 画面の幅

